apps/emqx/src/emqx_cm.erl

```diff
@@ -274,15 +282,20 @@ open_session(false, ClientInfo = #{clientid := ClientId}, ConnInfo) ->
                                      pendings => Pendings}};
                           {living, ConnMod, ChanPid, Session} ->
                               ok = emqx_session:resume(ClientInfo, Session),
-                              Session1 = emqx_persistent_session:persist( ClientInfo
-                                                                        , ConnInfo
-                                                                        , Session
-                                                                        ),
-                              Pendings = ConnMod:call(ChanPid, {takeover, 'end'}, ?T_TAKEOVER),
-                              register_channel(ClientId, Self, ConnInfo),
-                              {ok, #{session  => Session1,
-                                     present  => true,
-                                     pendings => Pendings}};
+                              case request_stepdown(
+                                     {takeover, 'end'},
+                                     ConnMod,
+                                     ChanPid) of
+                                  {ok, Pendings} ->
+                                      Session1 = emqx_persistent_session:persist(
+                                                   ClientInfo, ConnInfo, Session),
+                                      register_channel(ClientId, Self, ConnInfo),
+                                      {ok, #{session  => Session1,
+                                             present  => true,
+                                             pendings => Pendings}};
+                                  {error, _} ->
+                                      CreateSess()
```

Comment:
I'm not sure, if takeover-end fails, do we need to clean up some kind of resource, such as the already persistent session?

Reply:
what do you mean by already persistent session ?

Reply:
🤔I'm thinking if the request {takeover, 'end'} fails here, do I need to perform any cleanup action? For example, clean up the old session in the persistent store?

Reply:
For the on-disk sessions?
Don’t worry about it for now